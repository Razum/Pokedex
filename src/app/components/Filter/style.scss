.wrapper {
  position: relative;
  padding: 1rem;
}

.input {
  height: 3rem;
  line-height: 3rem;
  border: 1px solid darken(color(gray), 20);
  width: 100%;
  font-size: 1.5rem;
  padding: 0 1rem;
  transition: box-shadow 0.15s ease;

  &:focus {
    box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
  }
}