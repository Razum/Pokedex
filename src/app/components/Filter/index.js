import React from 'react'
import PropTypes from 'prop-types'
import {debounce} from 'lodash'

import style from './style.scss'

const FILTER_DELAY = 200

class Filter extends React.PureComponent {
  constructor () {
    super()
    this.onChangeHandle = this.onChangeHandle.bind(this)
    this.delayedFilter = debounce((filterTerm) => {
      this.props.callback(filterTerm)
    }, FILTER_DELAY)
  }
  onChangeHandle (event) {
    this.delayedFilter(event.target.value)
  }
  render () {
    return (
      <div className={style.wrapper}>
        <input type='text' className={style.input} onChange={this.onChangeHandle} />
      </div>
    )
  }
}

Filter.propTypes = {
  callback: PropTypes.func
}

export default Filter
