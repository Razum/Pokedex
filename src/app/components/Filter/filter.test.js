import React from 'react'
import Filter from './'
import renderer from 'react-test-renderer'

describe('Filter component', () => {
  test('should match snapshot', () => {
    const tree = renderer.create(
      <Filter />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
