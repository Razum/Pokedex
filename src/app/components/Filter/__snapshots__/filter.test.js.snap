// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should match snapshot 1`] = `
<div
  className="wrapper"
>
  <input
    className="input"
    onChange={[Function]}
    type="text"
  />
</div>
`;
